### PR TITLE
Incapsulate the xhr from fileUploader into the ajax utility

### PR DESCRIPTION
**Pull request by nightskylark**
- Added necessary functionality for the `ajax` utility
- Replaced `XMLHttpRequest` with `ajax` utility in the FileUploader

### CODE REVIEW

**Review by Aden-git — DISMISSED** [review]

File: js/core/utils/ajax.js [hunk 1]
```diff
@@ -2,8 +2,56 @@
...
+var isStatusSuccess = function(status) {
+    return 200 <= status && status < 300;
+};
+
 var sendRequest = function(options) {
-    return $.ajax.call($, options);
+    options.xhrFields = options.xhrFields || {};
+    options.xhrFields.responseType = options.responseType;
+
+    if(options.upload) {
```
Comment (timbset) [inline js/core/utils/ajax.js:13]: Use negative condition to decrease nesting level
```
if(!options.upload) {
    return $.ajax(options);
}

var xhr = ...
...
```

File: js/core/utils/ajax.js [hunk 2]
```diff
@@ -2,8 +2,56 @@
...
+        var xhr = new XMLHttpRequest();
+        var d = $.Deferred();
+        var result = d.promise();
+
+        xhr.open(options.method, options.url, true);
+        xhr.upload["onprogress"] = options.upload["onprogress"];
```
Comment (timbset) [inline js/core/utils/ajax.js:19]: Why don't you use dot notation?

File: js/ui/file_uploader.js [hunk 3]
```diff
@@ -1165,70 +1164,50 @@ var FileUploader = Editor.inherit({
...
-                var status = e.currentTarget.status;
+        ajax.sendRequest({
+            url: this.option("uploadUrl"),
+            method: this.option("uploadMethod"),
+            headers: this.option("uploadHeaders"),
+            beforeSend: function(xhr) {
+                file.request = xhr;
+            },
+            upload: {
+                "onprogress": function(e) {
```
Comment (timbset) [inline js/ui/file_uploader.js:1181]: Here is redudant quotes

**Review by timbset — CHANGES_REQUESTED** [review]

File: js/core/utils/ajax.js (hunk 1 above)
Comment (nightskylark) [reply js/core/utils/ajax.js:13]: This method will be rewritten much to get rid of $.ajax. But till then for the most scenarios, the $.ajax will be used. So, it`s better to see the upload scenario in the separated condition block.

File: js/core/utils/ajax.js (hunk 2 above)
Comment (nightskylark) [reply js/core/utils/ajax.js:19]: The "onprogress" is not valid word for the linter. I am not sure it is correct to add the linter exception for this words.

File: js/ui/file_uploader.js (hunk 3 above)
Comment (nightskylark) [reply js/ui/file_uploader.js:1181]: The same reason as previous.

**Review by timbset — APPROVED** [review]

**Review by ovchinnikov — APPROVED** [review]

**Pull request merged**